deploy/kubernetes/Makefile

```diff
@@ -1,7 +1,7 @@
 ALL=elastic-agent-standalone elastic-agent-managed
 
 BEAT_VERSION=$(shell grep '^:stack-version:' ../../version/docs/version.asciidoc | cut -c 17- )
-BRANCH_VERSION=$(shell grep '^:doc-branch:' ../../version/docs/version.asciidoc | cut -c 14- )
```

Comment:
It appears the `:stack-version` key here isn't being updated properly.

The one you want and is guarnateed to be right is https://github.com/elastic/elastic-agent/blob/99b14c8acc63e65c710dca19e2115200209ff314/version/version.go#L7

There are a couple of examples for parsing this you could reuse. For one example see https://github.com/elastic/elastic-agent/blob/99b14c8acc63e65c710dca19e2115200209ff314/.buildkite/scripts/common.sh#L16

You'd still have to drop the patch version.

Reply:
Ah I also now see why we were trying to use doc-branch because this needs to be `main` and not `8.12` for main.

I would suggest `git branch --show-current` but that will break when this is run on developer branches.

Reply:
I changed it to read from version.go

Both managed and standalone updated